cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -551,6 +553,13 @@ func createEmptyShard(args *indexArgs) error {
 	return builder.Finish()
 }
 
+func (s *Server) AddHandlers(mux *http.ServeMux) {
+	s.httpHandlersRegisteredOnce.Do(func() {
```

Comment:
why do you wrap with a once? There is no side effects from what I can tell, you are mutating the input mux only.